workflow/type.go

```diff
@@ -15,8 +25,46 @@ type task struct {
 	TaskKey      string
 }
 
+type filter struct {
```

Comment:
rename to eventFilter

Plus logic to match te events could be implemented directrly in the event package

Reply:
do you mean having something like
```
event.Match(filter)
```
in that case why not but we need to have some kind of standard filter. This filter is really specific to workflows only.

Also for the renaming, let's not worry too much now as this will move.
https://github.com/mesg-foundation/engine/blob/260f7fb35f07fca05a15b7d30ddd834770cb31c2/workflow/type.go#L8-L9

Reply:
I really like to have a dedicated filter package that event, execution, instance and services could implement ;)

Reply:
let's not have this dedicated filter now, also this means that the logic of filtering is inverted.

Instead of `filter.Match(resource)` we have `resource.Match(filter)`. This could have some undesired implications